spec.bs

```diff
@@ -8349,6 +8630,101 @@ scoring signals requests=] |entriesToBatch|:
+1. Let |compressionIdMap| be an empty [=map=], whose [=map/keys=] are [=tuples=] of ([=origin=],
+  [=origin=]) and [=map/values=] are integers.
+1. Let |renderUrlIdMap| be an empty [=map=], whose [=map/keys=] are [=URLs=] and [=map/values=] are [=tuples=]
+  of (interger, integer).
+1. Let |nextCompressionGroupId| be 0.
+1. [=map/For each=] |request| of |entriesToBatch|:
+  1. Let |partitionId| be an integer.
+  1. Set |joiningOrigin| to |group|'s [=interest group/joining origin=].
+  1. Let |ownerOrigin| be |request|'s [=trusted scoring signals request/owner origin=].
+  1. Let |mapKey| be a tuple of |ownerOrigin| and |joiningOrigin|.
+  1. If |compressionIdMap| does not [=map/contain=] |mapKey|:
+    1. Set |compressionIdMap|[|mapKey|] to |nextCompressionGroupId|.
+    1. Increase |nextCompressionGroupId| by 1.
+  1. Let |compressionGroupId| be |compressionIdMap|[|mapKey|].
+  1. If |compressionGroupMap| does not [=map/contain=] |compressionGroupId|:
+    1. Let |compressionGroupMap|[|compressionGroupId|] be an empty [=map=], whose [=map/keys=] are integers
+      and [=map/values=] are [=maps=].
+  1. Set |partitionId| to [=list/size=] of |compressionGroupMap|[|compressionGroupId|].
+  1. Set |renderUrlIdMap|[|request|'s [=trusted scoring signals request/render URL=]] to [=tuple=]
+    of |compressionGroupId| and |partitionId|.
+  1. Let |partition| be an empty [=map=], whose [=map/keys=] are [=strings=] and [=map/values=] are [=strings=] or
+      [=list=] of [=strings=].
+  1. Set |partition|["id"] to |partitionId|.
+  1. Let |namespace| be an empty [=map=], whose [=map/keys=] are [=strings=] and [=map/values=] are [=strings=] or
+      [=list=] of [=strings=].
+  1. Set |namespace|["render_url"] to [|request|'s [=trusted scoring signals request/render URL=]].
+  1. Set |namespace|["ad_component_render_urls"] to |request|'s [=trusted scoring signals request/ad
+    component URLs=].
+  1. Set |partition|["namespace"] to |namespace|.
+  1. Let |metadata| be an empty [=map=], whose [=map/keys=] and [=map/values=] are [=strings=].
+  1. Set |metadata|["experiment_group_id"] to |firstRequest|'s [=trusted scoring signals
+      request/seller experiment group id=].
+  1. Set |partition|["metadata"] to |metadata|.
+  1. Set |compressionGroupMap|[|compressionGroupId|][|partitionId|] to |partition|.
+1. [=map/For each=] |id| → |group| in |compressionGroupMap|:
+  1. Let |compressionGroup| be an empty [=map=], whose [=map/keys=] are [=strings=] and [=map/values=] are integers or
+    [=lists=] of [=maps=].
+  1. Set |compressionGroup|["compression_group_id"] to |id|.
+  1. Set |compressionGroup|["partitions"] to an empty [=list=].
+  1. [=list/For each=] |partition| in |group|'s [=map/values=]:
+    1. [=list/Append=] |partition| to |compressionGroup|["partitions"].
+  1. [=list/Append=] |compressionGroup| to |compressionGroups|.
+1. Let |metadata| be an empty [=map=], whose [=map/keys=] and [=map/values=] are [=strings=].

```

Comment:
so this |metadata| and the other |metadata| above are totally different things? If so, let's give them clearer names.

Reply:
Yeah this `metadata` is a top-level one, and the above one is for partition level. Change above one to `partitionMetadata`.